htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -371,3 +383,103 @@ def classifyText(self, token, reset=0):
                            implies no decision could be made.
     """
     raise NotImplementedError
+
+
+  def save(self, saveModelDir):
+    """ Save the model in the given directory via pickling.
+
+    @param saveModelDir (string)
+           Directory path for saving the model. This directory should
+           only be used to store a saved model. If the directory does not exist,
+           it will be created automatically and populated with model data. A
+           pre-existing directory will only be accepted if it contains previously
+           saved model data. If such a directory is given, the full contents of
+           the directory will be deleted and replaced with current model data.
+
+    Implementation note: Subclasses should override _serializeExtraData() to
+    save any data that cannot be pickled.
```

Comment:
> FWIW dill is able to serialize more data types than pickle, but I have not yet tried using it. And I doubt it will help with serializing the network API instances.

Yes, this is mostly for saving network API instances. The save/load API as specified is flexible enough to handle anything - we don't have to use pickle. I'll update the comment to make it a bit more generic.